### ISSUE TITLE
Add module and line number to log messages

### ISSUE THREAD

**Issue by turboc1208** [home-assistant/appdaemon#25]
This would be helpful in tracking down issues. (at least for me)

In appapi

At the top
`import inspect    
`
change log as follows
```

  def log(self, msg, level = "INFO"):
   msg=str(inspect.stack()[1][3]) + " (" + str(inspect.stack()[1][2])+") - " + msg
   ha.log(self._logger, level,msg, self.name)

```

output looks like this now

```
2017-01-23 14:56:00.921066 INFO bigdata: restart_Big_Data (44) - In restart_Big_Data
2017-01-23 14:56:00.941680 INFO bigdata: start_Big_Data (34) - called listen_state
2017-01-23 14:56:00.951857 INFO bigdata: start_Big_Data (36) - called listen_event
2017-01-23 14:56:01.066585 INFO calalarm: schedulealarm (111) - In schedulealarm - master
2017-01-23 14:56:01.077012 INFO calalarm: schedulealarm (113) - room=master, owner=['chip', 'susan']
2017-01-23 14:56:01.097669 INFO calalarm: schedulealarm (116) - owner=chip, room=master, meeting=wakeup
2017-01-23 14:56:01.128598 INFO bigdata: event_happened (39) - event=SPEAK_EVENT
2017-01-23 14:56:01.132161 INFO speak: handle_speak_event (45) - handling speak event SPEAK_EVENT text=ahlexa, Turn off stairway light switch language=en  priority=1
2017-01-23 14:56:01.139090 INFO calalarm: addalarm (68) - adding alarm
2017-01-23 14:56:01.149303 INFO calalarm: addalarm (71) - existing timer 2017-01-24 05:30:00, interval 0, kwargs={}
2017-01-23 14:56:01.158923 INFO calalarm: addalarm (76) - Duplicate alarm
2017-01-23 14:56:01.167875 INFO calalarm: schedulealarm (119) - alarm scheduled for 2017-01-24 05:30:00
2017-01-23 14:56:01.180028 INFO calalarm: schedulealarm (116) - owner=susan, room=master, meeting=

```

**Comment by acockburn** [comment]
Looks interesting but I will add a twist - I'll implement this so you can include something like "__line__" and "__module__" and they will get substituted into the log message - that way it's optional.

**Comment by turboc1208** [comment]
cool

**Comment by acockburn** [comment]
Available in the next release - 1.5.0.
